### PR TITLE
web-wallet: The listener for `keyChange` event does a pointless check

### LINKED ISSUE #2846: The listener for `keyChange` event does a pointless check
**Describe the bug**
The listener [here](https://github.com/dusk-network/rusk/blob/master/web-wallet/src/routes/(app)/dashboard/send/%2Bpage.svelte#L38) does its check on the event type rather than on its detail's type.

Also the [contract statuses list](zZgpWUqw4GAeZ3V56HVW1Df8LBwT5EEXgF985cwr1YEneT7aCc7Acp4NQ78BhoTjYYek4gCwMo8qPdpYcGoLXmQtVkcknPkiDG1paHDG6b6du2hLjvRQTZqQEaSJeg8MzCF) isn't updated after the event fires and the "shield more dusk" component shows always the shielded spendable as a value.

### PR DESCRIPTION
**Pull request by ascartabelli**
Resolves #2846

### CODE REVIEW

File: web-wallet/src/routes/(app)/dashboard/send/__tests__/page.spec.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
-import { afterEach, describe, expect, it, vi } from "vitest";
-import { cleanup, render } from "@testing-library/svelte";
-import Send from "../+page.svelte";
+import { afterAll, afterEach, describe, expect, it, vi } from "vitest";
+import { cleanup, fireEvent, render } from "@testing-library/svelte";
+
+import mockedWalletStore from "../../../../../__mocks__/mockedWalletStore";
```
Comment (nortonandreev) [inline web-wallet/src/routes/(app)/dashboard/send/__tests__/page.spec.js:4]: I think at this point it's worth creating an alias for the mocks folder?

File: web-wallet/src/routes/(app)/dashboard/send/+page.svelte [hunk 2]
```diff
@@ -9,36 +9,44 @@
...
-    },
-  ];
-  /* eslint-disable no-sequences, no-unused-expressions */
-  $: balance, (spendable = balance.shielded.spendable);
-  /* eslint-enable no-sequences, no-unused-expressions */
+  $: [spendable, statuses] = getContractInfo(spendableSource, balance);
 
   /**
-   * @param {{type:string}} event
+   * @param {CustomEvent<{ type: "account" | "address" | undefined}>} event
```
Comment (nortonandreev) [inline web-wallet/src/routes/(app)/dashboard/send/+page.svelte:43]: Wondering if the undefined makes sense? We shouldn't end up in that situation?

**Review by nortonandreev — APPROVED** [review]

File: web-wallet/src/routes/(app)/dashboard/send/+page.svelte (hunk 2 above)
Comment (ascartabelli) [reply web-wallet/src/routes/(app)/dashboard/send/+page.svelte:43]: When the address is empty, for example.

File: web-wallet/src/routes/(app)/dashboard/send/__tests__/page.spec.js (hunk 1 above)
Comment (ascartabelli) [reply web-wallet/src/routes/(app)/dashboard/send/__tests__/page.spec.js:4]: https://github.com/dusk-network/rusk/issues/2849

**Review by kieranhall — APPROVED** [review]

**Pull request merged**